jax/experimental/export/_export.py

```diff
@@ -307,6 +307,17 @@ def __str__(self):
     # do not want the entire serialized module to end up in locations.
     return f"Exported(fun_name={self.fun_name}, ...)"
 
+  def in_shardings_for_mesh(self,
+                            mesh: sharding.Mesh) -> Sequence[sharding.XLACompatibleSharding | None]:
+    """Create XLACompatibleSharding for the inputs.
+
+    `in_shardings` are HloShardings, which are serializable because they are
+    independent of the mesh and the device assignment.
+    This method turns them into XLACompatibleSharding for the given mesh.
+    """
+    return tuple(sharding.GSPMDSharding(mesh.devices, s) if s is not None else None
+                 for s in self.in_shardings)
```

Comment:
in_shardings is a weird name for HloSharding. Can you call it `self.in_hlo_shardings` please?

Reply:
Summarizing offline discussion: `Exported` is meant to be more general that JAX: it is a StableHLO module with metadata for how to call it, and it could in principle be generated by other frameworks. Thus, `in_shardings` here means HloShardings. Furthermore, there is a cost at this point to changing these public names.